inductiva/tasks/task.py

```diff
@@ -379,14 +382,31 @@ def download_outputs(
             rm_downloaded_zip_archive: Whether to remove the archive after
             uncompressing it. If uncompress is False, this argument is ignored.
         """
-        api_response = self._api.download_task_output(
-            path_params=self._get_path_params(),
-            query_params={
-                "filename": filenames or [],
-            },
-            stream=True,
-            skip_deserialization=True,
-        )
+        success = False
+        failed = 0
+
+        while not success:
+            try:
+                api_response = self._api.download_task_output(
+                    path_params=self._get_path_params(),
+                    query_params={
+                        "filename": filenames or [],
+                    },
+                    stream=True,
+                    skip_deserialization=True,
+                )
+                success = True
+            except exceptions.ApiException as e:
+                failed += 1
+                if failed > constants.TASK_DOWNLOAD_MAX_API_REQUESTS:
+                    raise RuntimeError(
+                        "Something went wrong while downloading the task "
+                        "outputs. Please try again later.\n"
+                        "Keep in mind that some outputs may take some time "
+                        "to be transfered from the resource to the bucket."
+                    ) from e
+                time.sleep(1)
```

Comment:
maybe instead:

```python
for _ in range(constants.TASK_DOWNLOAD_MAX_API_REQUESTS):
    try:
        api_response = self._api...
        return
    time.sleep(1)
raise RuntimeError(
                        "Something went wrong while downloading the task "
                        "outputs. Please try again later.\n"
                        "Keep in mind that some outputs may take some time "
                        "to be transfered from the resource to the bucket."
                    )
```

This way we don't need `failed += 1` and `if failed > constants....` statments